src/geometry/manifold/manifold-applyops-minkowski.cc

```diff
@@ -25,42 +26,36 @@ namespace ManifoldUtils {
  */
 std::shared_ptr<const Geometry> applyMinkowskiManifold(const Geometry::Geometries& children)
 {
-  using Hull_kernel = CGAL::Epick;
-  using Hull_Mesh = CGAL::Surface_mesh<CGAL::Point_3<Hull_kernel>>;
-  using Hull_Points = std::vector<Hull_kernel::Point_3>;
-  using Nef_kernel = CGAL_Kernel3;
+  using Hull_Point = linalg::vec<double, 3>;
+  using Hull_Points = std::vector<Hull_Point>;
   using Polyhedron = CGAL_Polyhedron;
 
-  auto polyhedronFromGeometry = [](const std::shared_ptr<const Geometry>& geom, bool *pIsConvexOut) -> std::shared_ptr<Polyhedron>
+  auto polyhedronFromGeometry = [](const std::shared_ptr<const Geometry>& geom, bool *pIsConvexOut) -> std::shared_ptr<Polyhedron> 
   {
-    auto ps = std::dynamic_pointer_cast<const PolySet>(geom);
-    if (ps) {
+    if (auto ps = dynamic_cast<const PolySet *>(geom.get())) {
       auto poly = std::make_shared<Polyhedron>();
       CGALUtils::createPolyhedronFromPolySet(*ps, *poly);
       if (pIsConvexOut) *pIsConvexOut = ps->isConvex();
       return poly;
+    } else if (auto mani = dynamic_cast<const ManifoldGeometry *>(geom.get())) {
+      auto poly = mani->toPolyhedron<Polyhedron>();
+      if (pIsConvexOut) *pIsConvexOut = CGALUtils::is_weakly_convex(*poly);
```

Comment:
Wondering how slow this conversion to CGAL + convex check is.

Reply:
Tbh I've tried all sorts of things to minimize conversions, including better convexity propagation in the manifold case, and avoiding to build a poly (adhoc visitation of subshell vertices), somehow can't manage any actual improvement on this PR on the benchmarks (**sighs**) --> cf. branch [manifold-hull2](https://github.com/ochafik/openscad/compare/manifold-hull...ochafik:openscad:manifold-hull2?expand=1).

Tempted to leave this as is for now.